gobblin-api/src/main/java/gobblin/configuration/WorkUnitState.java

```diff
@@ -171,6 +172,14 @@ public void setHighWaterMark(long value) {
   }
 
   @Override
+  public Properties getProperties() {
+    Properties props = new Properties();
+    props.putAll(this.workunit.getProperties());
+    props.putAll(super.getProperties());
```

Comment:
I suggest changing to `this.getProperties`. This make it clearer that it is adding properties belonging to this `WorkUnitState` instance.